app/controllers/devise/registrations_controller.rb

```diff
@@ -96,8 +96,8 @@ def update_resource(resource, params)
 
   # Build a devise resource passing in the session. Useful to move
   # temporary session data to the newly created user.
-  def build_resource(hash=nil)
-    self.resource = resource_class.new_with_session(hash || {}, session)
+  def build_resource(hash = {})
```

Comment:
This changes the behaviour. Now, if someone passes `build_resource(nil)`, we will send nil to the model, which is not what we want.